src/adaptors/in-memory-articles.ts

```diff
@@ -0,0 +1,39 @@
+import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+import Articles from '../articles';
+import ArticleNotFound from '../errors/article-not-found';
+import InvalidId from '../errors/invalid-id';
+
+export default class InMemoryArticles implements Articles {
+  private nodes: { [key: string]: JsonLdObj } = {};
+
+  async add(node: JsonLdObj): Promise<void> {
+    if (node['@id'] === undefined) {
+      throw new InvalidId();
```

Comment:
`ArticleHasNoId` or something? (The ID itself isn't invalid as it doesn't exist.)